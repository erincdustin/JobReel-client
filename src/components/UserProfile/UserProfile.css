@media screen and (min-width: 960px) {
  .UserProfile {
    border: 1px black solid;
    min-width: 300px;
    margin: 0 auto;
    padding: 5%;
    line-height: 150%;
    border-radius: 10px;
    text-align: center;
    margin: auto;
    background: #3a3846;
    color: white;
    grid-column-start: 6;
    grid-column-end: 9;
    box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  }
}

@media screen and (max-width: 960px) {
  .UserProfile {
    border: 1px black solid;
    min-width: 300px;
    margin: 0 auto;
    padding: 5%;
    line-height: 150%;
    border-radius: 10px;
    text-align: center;
    margin: auto;
    background: #3a3846;
    color: white;
    grid-column-start: 1;
    grid-column-end: 9;
    box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  }
}

