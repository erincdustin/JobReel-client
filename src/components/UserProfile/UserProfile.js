import React from 'react';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import Button from '../Button/Button';

class UserProfile extends React.Component {

  static getDerivedStateFromProps(props) {
    const { user } = props;
    console.log(user);
    return null;
  }

  render() {
    const { user } = this.props;
    const location = user.location ? user.location : <Button onClick={this.props.updateEditingProfile}>Add Location</Button>;
    const industry = user.industry ? user.industry : <Button onClick={this.props.updateEditingProfile}>Add Industry</Button>;
    const jobTitle = user.job_title ? user.job_title : <Button onClick={this.props.updateEditingProfile}>Add Job Title</Button>;
    return (
      <div className="UserProfile">
        <h3>{user.first_name} {user.last_name}</h3>
        <p>Username: {user.username}</p>
        <p>Email: {user.email}</p>
        <p>Location: {location}</p>
        <p>Industry: {industry}</p>
        <p>Job Title: {jobTitle}</p>
      </div>
    )
  }
}

export default UserProfile;