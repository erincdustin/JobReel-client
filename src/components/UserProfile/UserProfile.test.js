import React from 'react'
import ReactDOM from 'react-dom'
import UserProfile from './UserProfile'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const dummyData = {
  first_name: 'test',
  last_name: 'test',
  username: 'test',
  email: 'test',
  city: 'test',
  industry: 'test',
  job_title: 'test'
}

describe('UserProfile smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<UserProfile user={dummyData} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('UserProfile snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
        <UserProfile user={dummyData}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})