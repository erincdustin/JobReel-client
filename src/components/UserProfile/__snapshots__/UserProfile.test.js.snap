// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile snapshot test renders the UI as expected 1`] = `
<div
  className="UserProfile"
>
  <h3>
    test
     
    test
  </h3>
  <p>
    Username: 
    test
  </p>
  <p>
    Email: 
    test
  </p>
  <p>
    City: 
    test
  </p>
  <p>
    Industry: 
    test
  </p>
  <p>
    Job Title: 
    test
  </p>
  <ForwardRef>
    Edit
  </ForwardRef>
</div>
`;
